app/models/Photo.java

```diff
@@ -0,0 +1,215 @@
+    private Restaurant restaurant;
+
+    public Photo(Http.MultipartFormData.FilePart upload, int restaurantId) {
+        String fileExtension = FilenameUtils.getExtension(upload.getFilename());
+        name = UUID.randomUUID().toString() + "." + fileExtension;
+        file = upload.getFile();
+        this.restaurant = PersistenceManager.getRestaurantById(restaurantId);
+        isDefault = restaurant.getPhotos().size() == 0 ? true : false;
+    }
+
+	@JsonView(View.AllDetails.class)
+    public URL getUrl() throws MalformedURLException {
+        return new URL("https://s3.amazonaws.com/" + bucket + "/" + getNameOriginal());
+    }
+
+    @Override
+    public void delete() {
+        if (S3Plugin.amazonS3 == null) {
+            Logger.error("Could not delete because amazonS3 was null");
+            throw new RuntimeException("Could not delete");
+        }
+        else {
+            S3Plugin.amazonS3.deleteObject(bucket, getNameOriginal());
+            S3Plugin.amazonS3.deleteObject(bucket, getNameForMedium());
+            S3Plugin.amazonS3.deleteObject(bucket, getNameForThumbnail());
+            super.delete();
+        }
+    }
+
+    @Override
+    public void save() {
+        if (S3Plugin.amazonS3 == null) {
+            Logger.error("Could not save because amazonS3 was null");
+            throw new RuntimeException("Could not save");
+        }
+        else {
+            this.bucket = S3Plugin.s3Bucket;
+            super.save(); // assigns an photoId
+
+            uploadPhotoToS3(bucket, getNameOriginal(), this.file);
+            uploadPhotoToS3(bucket, getNameForThumbnail(), getThumbnail());
+            uploadPhotoToS3(bucket, getNameForMedium(), getMediumPhoto());
+        }
+    }
+
+    private void uploadPhotoToS3(String bucket, String name, File file){
+        PutObjectRequest putObjectRequest = new PutObjectRequest(bucket, name, file);
+        putObjectRequest.withCannedAcl(CannedAccessControlList.PublicRead); // public for all
+        S3Plugin.amazonS3.putObject(putObjectRequest); // upload file
+    }
+
+    private File getMediumPhoto(){
+        BufferedImage in = null;
+        File temp = new File(name);
+        try {
+            in = ImageIO.read(file);
+            if(in != null){
+                in = resize(in, 500, 500);
+            }
+            ImageIO.write(in, FilenameUtils.getExtension(name).toUpperCase(), temp);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return temp;
+    }
+
+    private File getThumbnail(){
+        BufferedImage in = null;
+        File temp = new File(name);
+        try {
+            in = ImageIO.read(file);
+            if(in != null){
+                in = resize(in, 50, 50);
+            }
+            ImageIO.write(in, FilenameUtils.getExtension(name).toUpperCase(), temp);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return temp;
+    }
```

Comment:
getMediumPhoto and getThumbnail are same, with the exception of size... you should create generic resize method and use input streams (as said in previous comments)